src/model/report.rs

```diff
@@ -12,4 +14,20 @@ impl Report {
             crash_list,
         }
     }
+
+    pub fn assign_operating_system_details(
+        &mut self,
+        operating_systems: Option<HashMap<String, Vec<OperatingSystemCount>>>,
+    ) {
+        if let Some(operating_systems_hash) = operating_systems {
+            for crash in &mut self.crash_list.crashes {
+                if let Some(group_id) = &crash.error_group_id {
+                    if let Some(oses) = operating_systems_hash.get(group_id) {
+                        let os_distribution: Vec<OperatingSystemCount> = oses.to_vec();
+                        crash.assign_operating_system_crash_distribution(os_distribution);
+                    }
+                }
+            }
+        }
```

Comment:
nitpick: I think you can use guard clause to get rid of all the nested ifs https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html

Reply:
Based on the example, I'm actually not  sure how to apply this?!
I'm using pairs and I do need to "unwrap" (for lack of better terminology) various optional types.

Reply:
I simplified this loop in some other way.